store/storage/rocksdb/db_test.go

```diff
@@ -71,6 +71,42 @@ func TestDatabase_CRUD(t *testing.T) {
 	require.Nil(t, val)
 }
 
+func TestDatabase_GetVersionedKey(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	// store a key at version 1
+	err = db.Set(storeKey1, 1, []byte("key"), []byte("value"))
+	require.NoError(t, err)
+
+	// assume chain progresses to version 10 w/o any changes to key
+	bz, err := db.Get(storeKey1, 10, []byte("key"))
+	require.NoError(t, err)
+	require.Equal(t, []byte("value"), bz)
+
+	ok, err := db.Has(storeKey1, 10, []byte("key"))
+	require.NoError(t, err)
+	require.True(t, ok)
+}
+
+func TestDatabase_VersionedKeys(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	for i := 1; i <= 100; i++ {
+		err := db.Set(storeKey1, uint64(i), []byte("key"), []byte(fmt.Sprintf("value%03d", i)))
+		require.NoError(t, err)
+	}
+
+	for i := 1; i <= 100; i++ {
+		bz, err := db.Get(storeKey1, uint64(i), []byte("key"))
+		require.NoError(t, err)
+		require.Equal(t, fmt.Sprintf("value%03d", i), string(bz))
```

Comment:
I think these test cases can be abstracted, work on the store interface, shared for all backends.

Reply:
WDYM exactly?

Reply:
Ohh, do you mean having a single test suite that tests over an interface that we pass each backend into? If so, I can do that yes!